clients/src/main/java/org/apache/kafka/clients/consumer/internals/ConsumerCoordinator.java

```diff
@@ -808,6 +808,31 @@ public boolean equals(Object o) {
         public int hashCode() {
             return partitionsPerTopic != null ? partitionsPerTopic.hashCode() : 0;
         }
+
+        public boolean topicsChanged(MetadataSnapshot that) {
+
+            if (partitionsPerTopic == null) {
+                return (that.partitionsPerTopic == null);
+            } else {
+                if (partitionsPerTopic.size() != that.partitionsPerTopic.size())
+                    return true;
+
+                for (String topic: partitionsPerTopic.keySet()) {
+                    if (that.partitionsPerTopic.containsKey(topic)) {
+                        if (that.partitionsPerTopic.get(topic) != null
+                                && this.partitionsPerTopic.get(topic) != null
+                                && !that.partitionsPerTopic.get(topic).equals(this.partitionsPerTopic.get(topic))) {
+
+                            return true;
+                        }
+                    } else {
+                        return true;
+                    }
+                }
+
+                return false;
```

Comment:
Hmm, are you doing anything different than `partitionsPerTopic == null : that.partitionsPerTopic == null : partitionsPerTopic.equals(that.partitionsPerTopic)`?